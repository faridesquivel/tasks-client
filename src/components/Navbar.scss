@import '../styles/colors';

.NavbarContainer {
    background-color: $primary;
    display: flex;
    padding: 10px;
    flex-direction: column;
}

.Title {
    color: aliceblue;
    font-size: 36px;
    padding: 15px;
}

.NavbarItemsContainer {
    display: flex;
    flex-direction: row;
}

.NavbarItem {
    color: aliceblue;
    padding: 15px;
}

.Link {
    text-decoration: none;
    font-size: 22px;
    color: $light;
}

.Link:hover {
    color: $secondary;
}

@media (max-width: 600px) {
    .NavbarContainer {
        border-radius: 0;
        width: 100vw;
    }
}