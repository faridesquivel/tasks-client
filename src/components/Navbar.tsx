import React from 'react';
import { useHistory } from 'react-router-dom';
import { 
    AppBar, 
    Toolbar, 
    IconButton, 
    Fab, 
    makeStyles, 
    createStyles, 
    Theme, 
    InputBase, 
    fade
} from '@material-ui/core';
import { ExitToApp, List } from '@material-ui/icons';
import AddIcon from '@material-ui/icons/Add';
import SearchIcon from '@material-ui/icons/Search';

const Navbar = ({ signOut, onFilterChange }: any) => {
    const classes = useStyles();
    const history = useHistory();
    return (
       <AppBar position="fixed" color="primary" className={classes.appBar}>
            <Toolbar>
                <IconButton title="List Button" onClick={() => history.push('tasks')} className={classes.leftIcon} edge="start" color="inherit" aria-label="open drawer">
                    <List />
                </IconButton>
                <div className={classes.search}>
                    <div className={classes.searchIcon}>
                        <SearchIcon />
                    </div>
                    <InputBase
                        placeholder="Searchâ€¦"
                        classes={{
                            root: classes.inputRoot,
                            input: classes.inputInput,
                        }}
                        onFocus={() => history.push('tasks')}
                        onChange={onFilterChange}
                        inputProps={{ 'aria-label': 'search' }}
                    />
                </div>
                <Fab title="Add Button" onClick={() => history.push('addTask')} color="secondary" aria-label="add" className={classes.fabButton}>
                    <AddIcon/>
                </Fab>
                <div className={classes.grow} />
                <IconButton title="Signout Button" onClick={signOut} className={classes.leftIcon}>
                    <ExitToApp className={classes.exitIcon}/>
                </IconButton>
            </Toolbar>
        </AppBar>
    );
};

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    text: {
      padding: theme.spacing(2, 2, 0),
    },
    exitIcon: {
      color: '#FFFFFF'
    },
    paper: {
      paddingBottom: 50,
    },
    list: {
      marginBottom: theme.spacing(2),
    },
    subheader: {
      backgroundColor: theme.palette.background.paper,
    },
    appBar: {
      top: '90%',
      bottom: 0,
      [theme.breakpoints.up('sm')]: {
          top: 'auto'
      }
    },
    grow: {
      flexGrow: 1,
    },
    fabButton: {
      position: 'absolute',
      zIndex: 1,
      top: -30,
      left: 0,
      right: 0,
      margin: '0 auto'
    },
    leftIcon: {
      marginTop: '25px',
      [theme.breakpoints.up('sm')]: {
        marginTop: 0
      },
    },
    search: {
      position: 'relative',
      borderRadius: theme.shape.borderRadius,
      backgroundColor: fade(theme.palette.common.white, 0.15),
      '&:hover': {
        backgroundColor: fade(theme.palette.common.white, 0.25),
      },
      marginLeft: 0,
      width: '100%',
      marginTop: '25px',
      [theme.breakpoints.up('sm')]: {
        marginLeft: theme.spacing(1),
        width: 'auto',
        marginTop: 0
      },
    },
    searchIcon: {
      padding: theme.spacing(0, 2),
      height: '100%',
      position: 'absolute',
      pointerEvents: 'none',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
    },
    inputRoot: {
      color: 'inherit',
    },
    inputInput: {
      padding: theme.spacing(1, 1, 1, 0),
      paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
      transition: theme.transitions.create('width'),
      width: '100%',
      [theme.breakpoints.up('sm')]: {
        width: '12ch',
        '&:focus': {
          width: '20ch',
        },
      },
    },
  }),
);

export default Navbar;