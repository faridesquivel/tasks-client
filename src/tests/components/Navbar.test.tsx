import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import Navbar from '../../components/Navbar';

describe('should render navbar', () => {
    it('should define navbar correctly', () => {
        const { getByText } = render(
            <Navbar />
        );   
        expect(getByText).toBeDefined()
    });
    it('should render navbar correctly', () => {
        render(
            <Navbar />
        ); 
    });
    it('should match navbar correctly', () => {
        const { baseElement } = render(
            <Navbar />
        ); 
        expect(baseElement).toMatchSnapshot()
    });
    it('should render list button correctly', () => {
        const { getByTitle } = render(
            <Navbar />
        );   
        expect(getByTitle('List Button')).toBeInTheDocument()
    });
    it('should render searchbar input correctly', () => {
        const { getByPlaceholderText } = render(
            <Navbar />
        );   
        expect(getByPlaceholderText('Searchâ€¦')).toBeInTheDocument()
    });
    it('should render add button correctly', () => {
        const { getByTitle } = render(
            <Navbar />
        );   
        expect(getByTitle('Add Button')).toBeInTheDocument()
    });
    it('should render signout button correctly', () => {
        const { getByTitle } = render(
            <Navbar />
        );   
        expect(getByTitle('Signout Button')).toBeInTheDocument()
    });
});