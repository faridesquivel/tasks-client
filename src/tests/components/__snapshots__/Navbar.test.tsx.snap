// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render navbar should match navbar correctly 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-32 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-leftIcon-35 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          title="List Button"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="makeStyles-search-36"
        >
          <div
            class="makeStyles-searchIcon-37"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root makeStyles-inputRoot-38"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input makeStyles-inputInput-39"
              placeholder="Searchâ€¦"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          aria-label="add"
          class="MuiButtonBase-root MuiFab-root makeStyles-fabButton-34 MuiFab-secondary"
          tabindex="0"
          title="Add Button"
          type="button"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="makeStyles-grow-33"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root makeStyles-leftIcon-35"
          tabindex="0"
          title="Signout Button"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-exitIcon-28"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
</body>
`;
